src/features/surveys/features/SurveyResults/components/IndividualResults/IndividualResults.tsx

```diff
@@ -1,5 +1,152 @@
+    const newValue = parseInt(e.target.value) - 1;
+    if (newValue >= 0 && newValue < totalAnswers) {
+      setCurrentIndex(newValue);
+    }
+  };
+
+  if (totalAnswers <= 0) {
+    return <></>;
+  }
+
+  return (
+    <div className="mt-8">
+      <div className="my-5 flex flex-row items-center justify-start gap-x-1">
+        <Button
+          title={'Previous'}
+          onClick={handlePrevious}
+          className="grow px-2 sm:grow-0"
+          variant={ButtonVariant.FLAT}
+          icon={<ChevronLeftIcon className="h-5 w-5" />}
+          disabled={currentIndex === 0}
+        />
+
+        <div className="flex flex-row items-center gap-x-2">
+          <div>
+            <Input
+              className="my-0 max-w-[100px]"
+              name="survey-title"
+              placeholder={'1'}
+              value={currentIndex + 1}
+              type="number"
+              error={''}
+              max={totalAnswers}
+              min={1}
+              step={1}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                handleInputChange(e)
+              }
+              inputSize={InputSize.SMALL}
+            />
+          </div>
+          <p>of</p>
+          <p>{surveyData?.answers.length || 0}</p>
+        </div>
+
+        <Button
+          title={'Next'}
+          onClick={handleNext}
+          className="grow px-2 sm:grow-0"
+          variant={ButtonVariant.FLAT}
+          icon={<ChevronRightIcon className="h-5 w-5" />}
+          disabled={currentIndex === totalAnswers - 1}
+        />
+      </div>
+
+      {Object.keys(mappedAnswersData).map((key, index) => {
+        const questionData = surveyData?.questions[index];
+        const question = mappedAnswersData[key].question;
+        const questionType = mappedAnswersData[key].questionType;
+        const answer = mappedAnswersData[key].answers[currentIndex].answer;
+
+        return (
+          <div
+            className="mb-3 rounded-md border bg-white p-4 text-start shadow"
+            key={index}
+          >
+            <h2 className="mb-4 text-lg font-semibold">{question}</h2>
+            {questionType === QuestionType.INPUT && <p>{answer}</p>}
+
+            <div className="flex flex-wrap justify-start gap-x-2">
+              {questionType === QuestionType.RATE &&
+                [...Array(5)].map((_, index) => (
+                  <StarComponent
+                    key={index}
+                    classNames={clsx(
+                      answer && index < +answer
+                        ? 'text-yellow-400'
+                        : 'text-gray-300'
+                    )}
+                    onMouseEnter={() => {
+                      console.log('OK');
+                    }}
+                    onMouseLeave={() => {
+                      console.log('OK');
+                    }}
+                    onClick={() => {
+                      console.log('OK');
+                    }}
```

Comment:
Please make all callbacks for each question type optional, and just don't pass it 😊

Reply:
Thanks for looking at my request. Sure, I will make all question type callbacks optional and pass only when needed. Let me know if there's anything else!